src/Microsoft.AspNetCore.SignalR.Core/HubOptions.cs

```diff
@@ -24,5 +25,7 @@ public class HubOptions
         /// functionality because of the polling mechanism. 
         /// </remarks>
         public TimeSpan KeepAliveInterval { get; set; } = DefaultKeepAliveInterval;
+
+        public List<string> SupportedProtocols { get; set; } = new List<string> { DefaultProtocols.JSONProtocol, DefaultProtocols.MsgPackProtocol};
```

Comment:
I was waiting for this to come up. I think by default you support all available transports. That is, not specified means all. There also needs to be validation like what if you clear all? What does that mean?

Reply:
Could we somehow grab all available transports when HubOptions is created and populate the list before running the users action on it?

Reply:
We have nested generics so I'm not sure how well it will work. You'd need to get all the configured IHubProtocol implementations inside of an `IConfigureOptions<HubOptions<THub>>` which makes it a bit difficult as there's no explicit call to register a hub.

Reply:
Other fun things: MsgPack is in a completely separate assembly that isn't available in `AspNetCore.All` currently.